File: csharp/sieve/Sieve.cs
using System;
using System.Linq;

public static class Sieve
{
    public static int[] Primes(int limit)
    {
        var smallestPrime = 2;

        if (limit < smallestPrime) { throw new ArgumentOutOfRangeException(); }

        var numbers = Enumerable.Range(smallestPrime, limit - 1).ToList();

        for(var i= smallestPrime; i<=Math.Sqrt(limit); i++)
        {
            var j = i*i;

            while(j <= limit)
            {
                numbers.Remove(j);
                j += i;
            }
        }

        return numbers.ToArray();
    }
}

File: go/sieve/sieve.go
package sieve

func Sieve(limit int) []int {
	if limit < 2 {
		return []int{}
	} else if limit == 2 {
		return []int{2}
	}

	mapNumberSequences := make([]bool, limit+1)
	primes := make([]int, limit/2)
	primeIndex := 0

	for number := 2; number <= limit; number++ {
		if !mapNumberSequences[number] {
			primes[primeIndex] = number
			primeIndex++

			for i := number + number; i <= limit; i += number {
				mapNumberSequences[i] = true
			}
		}

	}

	return primes[:primeIndex]
}
